entity-service-impl/src/test/java/org/hypertrace/entity/query/service/EntityQueryServiceImplTest.java

```diff
@@ -177,6 +197,106 @@ public void updateSuccess() throws Exception {
             eq(new JSONDocument(DocStoreJsonFormat.printer().print(newStatus))));
   }
 
+  @Test
+  public void testExecute_noTenantId() throws Exception {
+    StreamObserver<ResultSetChunk> mockResponseObserver = mock(StreamObserver.class);
+
+    Context.current()
+        .withValue(RequestContext.CURRENT, mock(RequestContext.class))
+        .call(
+            () -> {
+              EntityQueryServiceImpl eqs =
+                  new EntityQueryServiceImpl(mockEntitiesCollection(), attributeFqnMaps);
+
+              eqs.execute(null, mockResponseObserver);
+
+              verify(mockResponseObserver, times(1))
+                  .onError(argThat(
+                      new ExceptionMessageMatcher("Tenant id is missing in the request.")));
+              return null;
+            });
+  }
+
+  @Test
+  public void testExecute_success() throws Exception {
+    Collection mockEntitiesCollection = mock(Collection.class);
+    Entity ENTITY =
+        Entity.newBuilder()
+            .setTenantId("tenant-1")
+            .setEntityType(TEST_ENTITY_TYPE)
+            .setEntityId(UUID.randomUUID().toString())
+            .setEntityName("Test entity")
+            .putAttributes(
+                EDS_COLUMN_NAME1,
+                AttributeValue.newBuilder().setValue(
+                    org.hypertrace.entity.data.service.v1.Value.newBuilder().setString("foo1")).build())
+            .build();
+
+    List<Document> docs = java.util.Collections.singletonList(new JSONDocument(JsonFormat.printer().print(ENTITY)));
+    when(mockEntitiesCollection.search(any())).thenReturn(docs.iterator());
+    EntityQueryRequest request = EntityQueryRequest.newBuilder()
+        .setEntityType(TEST_ENTITY_TYPE)
+        .addOrderBy(
+            OrderByExpression.newBuilder()
+                .setExpression(
+                    Expression.newBuilder().setColumnIdentifier(
+                        ColumnIdentifier.newBuilder()
+                            .setColumnName(EQS_COLUMN_NAME1)
+                            .build()))
+        ).build();
+    StreamObserver<ResultSetChunk> mockResponseObserver = mock(StreamObserver.class);
+    Context.current()
+        .withValue(RequestContext.CURRENT, mockRequestContextWithTenantId())
+        .call(
+            () -> {
+              EntityQueryServiceImpl eqs =
+                  new EntityQueryServiceImpl(mockEntitiesCollection, attributeFqnMaps);
+
+              eqs.execute(request, mockResponseObserver);
+              return null;
+            });
+
+    verify(mockEntitiesCollection, times(1)).search(any());
+    verify(mockResponseObserver, times(1)).onNext(any());
```

Comment:
would consider adding one more document above, so we can check onNext called multiple times say ` times(2)` if we have two documents.
```
 List<Document> docs = java.util.Collections.singletonList(new JSONDocument(JsonFormat.printer().print(ENTITY)));
```